### PR TITLE
Update search and phone search for API v3

### PR DESCRIPTION
**Pull request by SSheldon**
(Based off #48)

This updated `YLPSearch` and `YLPQuery` from their API v2 format to API v3. Along the way I removed `YLPPhoneSearch`, since it was an identical copy of `YLPSearch`.

After this, all the tests can be re-enabled. I fixed them and updated the mock responses to match the new formats.

### CODE REVIEW

**Comment by SSheldon** [comment]
Until #48 is merged, its contents will also show up in this diff. To view just the search & phone search changes, you can select a range of the diff like https://github.com/Yelp/yelp-ios/pull/50/files/45554ab..6530847

File: Classes/Client/YLPClient+PhoneSearch.h [hunk 1]
```diff
@@ -7,22 +7,17 @@
...
 
 @interface YLPClient (PhoneSearch)
 
 - (void)businessWithPhoneNumber:(NSString *)phoneNumber
-                       countryCode:(nullable NSString *)countryCode
-                          category:(nullable NSString *)category
-                 completionHandler:(YLPPhoneSearchCompletionHandler)completionHandler;
-
-- (void)businessWithPhoneNumber:(NSString *)phoneNumber
                  completionHandler:(YLPPhoneSearchCompletionHandler)completionHandler;
```
Comment (thibaudrobelain) [inline Classes/Client/YLPClient+PhoneSearch.h]: indentation

File: Classes/Client/YLPClient+PhoneSearch.m [hunk 2]
```diff
@@ -8,53 +8,28 @@
...
 - (NSURLRequest *)businessRequestWithParams:(NSDictionary *)params {
-
-    
-    NSString *phoneSearchPath = @"/v2/phone_search/";
+    NSString *phoneSearchPath = @"/v3/businesses/search/phone";
     return [self requestWithPath:phoneSearchPath params:params];
 }
 
 - (void)businessWithPhoneNumber:(NSString *)phoneNumber
                  completionHandler:(YLPPhoneSearchCompletionHandler)completionHandler {
```
Comment (thibaudrobelain) [inline Classes/Client/YLPClient+PhoneSearch.m]: nit:indentation

File: Classes/Client/YLPClient+Search.m [hunk 3]
```diff
@@ -20,69 +18,45 @@ @implementation YLPClient (Search)
 
 - (void)searchWithLocation:(NSString *)location
             completionHandler:(YLPSearchCompletionHandler)completionHandler {
```
Comment (thibaudrobelain) [inline Classes/Client/YLPClient+Search.m]: nit: indentation

File: Classes/Client/YLPClient+PhoneSearch.m [hunk 4]
```diff
@@ -8,53 +8,28 @@
...
     
     [self queryWithRequest:req completionHandler:^(NSDictionary *responseDict, NSError *error) {
         if (error) {
             completionHandler(nil, error);
         } else {
-            YLPPhoneSearch *phoneSearch = [[YLPPhoneSearch alloc] initWithDictionary:responseDict];
-            completionHandler(phoneSearch, nil);
+            YLPSearch *search = [[YLPSearch alloc] initWithDictionary:responseDict];
+            completionHandler(search, nil);
         }
```
Comment (thibaudrobelain) [inline Classes/Client/YLPClient+PhoneSearch.m]: nit: remove

File: YelpAPITests/Client/YLPPhoneSearchClientTestCase.m [hunk 5]
```diff
@@ -59,60 +49,16 @@ - (void)testAttributesSetOnPhoneSearch{
...
-        return [request.URL.host isEqualToString:kYLPAPIHost];
-    } withStubResponse:^OHHTTPStubsResponse*(NSURLRequest *request) {
-        return [OHHTTPStubsResponse responseWithFileAtPath:OHPathForFile(self.minimalResource, self.class) statusCode:200 headers:@{@"Content-Type":@"application/json"}];
-    }];
-    
-    [self.client businessWithPhoneNumber:@"4151231234" completionHandler:^(YLPPhoneSearch *phoneSearchResults, NSError *error) {
-        YLPRegion *actualRegion = phoneSearchResults.region;
-        XCTAssertNil(actualRegion);
+        XCTAssertEqualObjects([actualBusinesses[0] class], [YLPBusiness class]);
         [expectation fulfill];
```
Comment (thibaudrobelain) [inline YelpAPITests/Client/YLPPhoneSearchClientTestCase.m]: nit: remove

**Comment by thibaudrobelain** [comment]
@SSheldon I had started writing comments about indentation but it seems like every method whose name is on at least two lines has this issue. Feel free to take a look around and remove these.
Otherwise it looks good to me!

**Pull request merged**